Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Snapping/snap_vertices.h

```diff
@@ -955,8 +955,19 @@ std::size_t snap_vertices_two_way(const HalfedgeRange_A& halfedge_range_A,
     {
       if(is_second_mesh_fixed)
       {
+        const Point& new_p = get(vpm_B, vb);
         for(const halfedge_descriptor ha : vs_a)
-          put(vpm_A, target(ha, tm_A), get(vpm_B, vb));
+        {
+          bool skip = false;
+          for (halfedge_descriptor haa : halfedges_around_target(ha, tm_A))
+            if (!is_border(haa,tm_A) && collinear(get(vpm_A, source(haa,tm_A)), new_p, get(vpm_A, target(next(haa,tm_A),tm_A))))
+            {
+              skip=true;
+              break;
+            }
+          if (!skip)
```

Comment:
same remark about inversed faces and applying moves while at least one within the tuple is prevented

Reply:
this seems to have been addressed